### PR TITLE
Implement weather data for HistoricalDataProvider and Azure Table consumer

### PR DESCRIPTION
**Pull request by henrymercer**

### CODE REVIEW

File: src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntityColumns.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * The number of minutes since midnight
+     * e.g. 00:34 is 34
+     */
+    String MINUTES = "Minutes";
+
+    /**
+     * The timestamp for the data point
+     */
+    String SEARCH_TIMESTAMP = "SearchTimestamp";
+}
```
Comment (henrymercer) [inline src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntityColumns.java]: Add rest of the columns to this interface

File: src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/azureconnectors/TableDoesNotExistException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package uk.ac.cam.alpha2017.airqualityradar.backend.data.azureconnectors;
+
+public class TableDoesNotExistException extends Exception {
```
Comment (jirkalhotka) [inline src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/azureconnectors/TableDoesNotExistException.java:3]: Doesn't Azure have such an exception itself?

File: src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntity.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import uk.ac.cam.alpha2017.airqualityradar.backend.models.WeatherDataPoint;
+import uk.ac.cam.alpha2017.airqualityradar.backend.models.measurements.NOxMeasurement;
+import uk.ac.cam.alpha2017.airqualityradar.backend.models.measurements.PM10Measurement;
+import uk.ac.cam.alpha2017.airqualityradar.backend.models.measurements.PM25Measurement;
+
+import java.text.ParseException;
+import java.util.Calendar;
+
+/**
+ * Creates a DataRowEntity which should be instantiated by reflection - that's why data fields don't follow
```
Comment (jirkalhotka) [inline src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntity.java]: Comment no longer makes sense

File: src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntityColumns.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package uk.ac.cam.alpha2017.airqualityradar.backend.data.entities;
+
+public interface DataRowEntityColumns {
```
Comment (jirkalhotka) [inline src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntityColumns.java]: Can we make this an enum?

Comment (henrymercer) [reply src/main/java/uk/ac/cam/alpha2017/airqualityradar/backend/data/entities/DataRowEntityColumns.java]: This class is used effectively to store the string values of the column names.  If we made it into an enum, I think we'd need to add a method to give the string value for each column enum.  So I don't think it's possible to make it into an enum without making it more complicated.

**Review by jirkalhotka — APPROVED** [review]

**Pull request merged**